src/basic/sysctl-util.c

```diff
@@ -58,6 +58,30 @@ int sysctl_write(const char *property, const char *value) {
 
         log_debug("Setting '%s' to '%s'", p, value);
 
+        if (shadow) {
+                _cleanup_free_ char *k = NULL, *v = NULL, *cur_k = NULL, *cur_v = NULL;
+                int r;
+
+                k = strdup(p);
+                if (!k)
+                        return -ENOMEM;
+
+                v = strdup(value);
+                if (!v)
+                        return -ENOMEM;
+
+                cur_k = hashmap_get2(*shadow, k, (void**)&cur_v);
+                if (cur_k)
+                        hashmap_remove(*shadow, cur_k);
+
+                r = hashmap_ensure_replace(shadow, &path_hash_ops_free_free, k, v);
```

Comment:
maybe use hashmap_ensure_put() here, after all you already removed the entry above